base/applications/taskmgr/graphctl.c

```diff
@@ -407,6 +419,52 @@ void GraphCtrl_Paint(TGraphCtrl* this, HWND hWnd, HDC dc)
     DeleteDC(memDC);
 }
 
+void GraphCtrl_ShiftGrid(TGraphCtrl* this)
+{
+    RECT rectCleanUp;
+    HPEN oldPen, solidPen;
+    int i;
+
+    if (this->m_dcGrid == NULL)
+        return;
+
+    solidPen = CreatePen(PS_SOLID, 0, this->m_crGridColor);
+
+    /*  BitBlt it to itself to scroll */
+    BitBlt(this->m_dcGrid, this->m_rectPlot.left, this->m_rectPlot.top + 1,
+           this->m_nPlotWidth, this->m_nPlotHeight, this->m_dcGrid,
+           this->m_rectPlot.left + this->m_nShiftPixels, this->m_rectPlot.top + 1,
+           SRCCOPY);
```

Comment:
```suggestion
    BitBlt(this->m_dcGrid, this->m_rectPlot.left, this->m_rectPlot.top + 1,
           this->m_nPlotWidth, this->m_nPlotHeight,
           this->m_dcGrid, this->m_rectPlot.left + this->m_nShiftPixels, this->m_rectPlot.top + 1,
           SRCCOPY);
```
Then, I know it's copy-pasted, but:
Why `top + 1`? Why (full) `m_nPlotWidth` and `m_nPlotHeight`? ..

Reply:
I'm not sure what to do with this one.

Reply:
Why is `+ 1` needed/correct? Why does the height not match it?
Why not something like `this->m_nPlotWidth - this->m_nShiftPixels`?
Aren't `left` and `top` simply `0`?

Reply:
`top` is -1, because when you're drawing the grid, the top horizontal line is not visible, so the original author used this trick: set `top` to -1, and you draw the horizontal line in the for loop, the first line will be outside the visible area and thus not drawn. Kind of a dirty trick by the original author.